src/init/variables/VarResolver.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable tsdoc/syntax */
+import yargs from 'yargs';
+import { createErrorMessage } from '../..';
+import { AsyncHandler } from '../../util/handlers/AsyncHandler';
+import { modulePathPlaceholder } from '../../util/PathUtil';
+import type { VarComputer } from './VarComputer';
+
+const defaultConfig = `${modulePathPlaceholder}config/default.json`;
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+export type YargsArgOptions = Record<string, yargs.Options>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function setupYargvWithBaseArgs(yargv: yargs.Argv) {
+  // Values from componentsjs LogLevel type
+  const logLevels = [ 'error', 'warn', 'info', 'verbose', 'debug', 'silly' ];
+  return yargv.options({
+    /**
+   * CLI options needed for performing meta-process of app initialization.
+   * These options doesn't contribute to components-js vars normally.
+   */
+    config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+    loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true, choices: logLevels },
+    mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
+    varConfig: { type: 'string', alias: 'v', default: defaultVarConfig, requiresArg: true },
+  });
+}
+
+export interface YargvOptions {
+  usage?: string;
+  strictMode?: boolean;
+  // @see http://yargs.js.org/docs/#api-reference-envprefix
+  loadFromEnv?: boolean;
+  envVarPrefix?: string;
+}
+
+export type VarRecord = Record<string, unknown>;
+
+export class VarResolver extends AsyncHandler<string[], VarRecord> {
+  protected readonly yargsArgOptions: YargsArgOptions;
+  protected readonly yargvOptions: YargvOptions;
+  protected readonly varComputers: Record<string, VarComputer>;
+
+  /**
+   * @param yargsArgOptions - record of option to it's yargs opt config. @range {json}
+   * @param yargvOptions - options to configure yargv. @range {json}
+   * @param varComputers  - record of componentsjs var-iri to VarComputer.
```

Comment:
```suggestion
   * @param parameters - Configuration of all supported parameters. @range {json}
   * @param extractors - Specifies for each variable URI how to extract its value from runtime parameter arguments.
   * @param options - Options for CLI parameter parsing.
```

Reply:
Here, extraction is only one case, when passed args directly corresponds to variable. In many cases we have to `compute` using passed args. like resolving asset paths, determining base url etc. hence they may not be just extractors?